core_impl/src/main/java/com/google/instrumentation/stats/StatsManagerImpl.java

```diff
@@ -81,6 +82,35 @@ StatsContextFactoryImpl getStatsContextFactory() {
     return statsContextFactory;
   }
 
+  /**
+   * Records a set of measurements with a set of tags.
+   *
+   * @param tags the tags associated with the measurements
+   * @param measurementValues the measurements to record
+   */
+  void record(StatsContextImpl tags, MeasurementMap measurementValues) {
+    queue.enqueue(new StatsEvent(measurementDescriptorToViewMap, tags.tags, measurementValues));
+  }
+
+  // An EventQueue entry that records the stats from one call to StatsManager.record(...).
+  private static final class StatsEvent implements EventQueueEntry {
+    private final Map<String, String> tags;
+    private final MeasurementMap stats;
+    private final MeasurementDescriptorToViewMap statsMap;
```

Comment:
I'm not sure whether we would like to use `StatsEvent` for `registerView()` and `getView()` as well. If that's the case, it would be better to store a reference to `StatsManagerImpl` instead of just `MeasurementDescriptorToViewMap`.